### PR TITLE
Quick Improvements 2019-04-11

### PR DESCRIPTION
**Pull request by daltonkell**
Addresses a few quick issues related to comparing data values to metadata, message improvements.

acdd.py:
  - check_lat_extents(), check_lon_extents(), and _check_total_z_extents() -- all tried to compare
    data values to metadata values, which may have been passed as strings. numpy copmarision would
    subsequently fail; fixed by casting the potential string values as floats (Issue #639)
  - check_coverage_content_type() was outputting the string "coverage_content" instead of
    "coverage_content_type"; inserted the correct string (Issue #637)

cf.py:
  - _check_formula_terms() was outputting a message about missing variables, but did not include
    the variables it was referring to. Fixed by adding a format field. (Issue #636)
  - check_grid_coordinates() message was a little ambiguous -- added some context, as well
    as a message for variables with missing coordinate variables (Issue #638)

### CODE REVIEW

**Comment by daltonkell** [comment]
Travis failure seems to be `conda`-related:
```
Fetching packages ...
conda-env-2.6. 100% |################################| Time: 0:00:00 724.30 kB/s
libffi-3.2.1-1 100% |################################| Time: 0:00:00   3.70 MB/s
sqlite-3.13.0- 100% |################################| Time: 0:00:00  48.24 MB/s
python-2.7.13- 100% |################################| Time: 0:00:00  65.10 MB/s
asn1crypto-0.2 100% |################################| Time: 0:00:00   5.19 MB/s
enum34-1.1.6-p 100% |################################| Time: 0:00:00   4.06 MB/s
idna-2.6-py27_ 100% |################################| Time: 0:00:00   5.48 MB/s
ipaddress-1.0. 100% |################################| Time: 0:00:00  27.48 MB/s
pycparser-2.18 100% |################################| Time: 0:00:00   5.81 MB/s
pyparsing-2.2. 100% |################################| Time: 0:00:00   4.90 MB/s
requests-2.12. 100% |################################| Time: 0:00:00  15.12 MB/s
ruamel_yaml-0. 100% |################################| Time: 0:00:00   9.72 MB/s
six-1.10.0-py2 100% |################################| Time: 0:00:00   9.02 MB/s
cffi-1.10.0-py 100% |################################| Time: 0:00:00  10.26 MB/s
packaging-16.8 100% |################################| Time: 0:00:00  17.22 MB/s
cryptography-1 100% |################################| Time: 0:00:00  15.28 MB/s
pyopenssl-17.0 100% |################################| Time: 0:00:00   4.17 MB/s
conda-4.3.30-p 100% |################################| Time: 0:00:00  11.57 MB/s
Extracting packages ...
[      COMPLETE      ]|###################################################| 100%
Unlinking packages ...
[      COMPLETE      ]|###################################################| 100%
Linking packages ...
[      COMPLETE      ]|###################################################| 100%
0.29s$ conda config --remove channels defaults --force
0.27s$ conda config --add channels conda-forge --force
0.02s$ python gen_conda_requirements.py --groups requirements.txt --py${PY:0:1} > conda-requirements.txt;
31.50s$ conda create --name TEST python=$PY --file conda-requirements.txt --file test_requirements.txt
Fetching package metadata .....
Solving package specifications: 
PackageNotFoundError: Packages missing in current channels:
            
  - python 3.7* -> libgcc-ng >=7.3.0
  - python 3.7* -> libstdcxx-ng >=7.3.0
  - python 3.7* -> openssl >=1.0.2p,<1.0.3a
  - python 3.7* -> sqlite >=3.25.2,<4.0a0
  - python 3.7* -> xz >=5.2.4,<5.3.0a0
We have searched for the packages in the following channels:
            
  - https://conda.anaconda.org/conda-forge/linux-64
  - https://conda.anaconda.org/conda-forge/noarch
            
The command "conda create --name TEST python=$PY --file conda-requirements.txt --file test_requirements.txt" failed and exited with 1 during .
Your build has been stopped.
```

File: compliance_checker/acdd.py [hunk 1]
```diff
@@ -226,8 +226,8 @@ def check_lat_extents(self, ds):
         if not (hasattr(ds, 'geospatial_lat_min') and hasattr(ds, 'geospatial_lat_max')):
             return
 
-        lat_min = ds.geospatial_lat_min
-        lat_max = ds.geospatial_lat_max
+        lat_min = float(ds.geospatial_lat_min)
+        lat_max = float(ds.geospatial_lat_max)
```
Comment (benjwadams) [inline compliance_checker/acdd.py]: Could you add exception handlers to these float casts?  Sometimes there are values that can't be cast to float.

Comment (benjwadams) [reply compliance_checker/acdd.py]: i.e. `except ValueError:`

**Comment by daltonkell** [comment]
@benjwadams, the changes you asked for are in the most recent commit

**Pull request merged**